tests/pld_budget_accounting_test.py

```diff
@@ -0,0 +1,58 @@
+from pipeline_dp.aggregate_params import NoiseKind
+import unittest
+from pipeline_dp.pld_budget_accounting import PLDBudgetAccountant
+
+
+class PLDBudgetAccountingTest(unittest.TestCase):
+
+    def test_lazy_budget_assertion_error(self):
+        accountant = PLDBudgetAccountant(total_epsilon=3, total_delta=1e-5)
+        budget = accountant.request_budget(NoiseKind.LAPLACE)
+        self.assertIsNotNone(budget)
+        self.assertRaises(AssertionError, getattr, budget, "noise_parameter")
+
+    def test_invalid_epsilon_none_delta(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=0)
+
+    def test_invalid_epsilon_valid_delta(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=0, total_delta=1e-5)
+
+    def test_invalid_delta_none_epsilon(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_delta=-1e-5)
+
+    def test_invalid_delta_valid_epsilon(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=1, total_delta=-1e-5)
+
+    def test_invalid_none_epsilon_none_delta(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant()
+
+    def test_invalid_noise_range(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(noise=100, total_epsilon=1, total_delta=1e-5)
+
+    def test_request_budget_multiple(self):
+        accountant = PLDBudgetAccountant(total_epsilon=3, total_delta=1e-5)
+        accountant.request_budget(NoiseKind.LAPLACE)
+        accountant.request_budget(NoiseKind.GAUSSIAN)
+        self.assertEqual(1, len(accountant._requested_laplace_budgets))
+        self.assertEqual(1, len(accountant._requested_gaussian_budgets))
+
+    def test_optimize_noise_laplace(self):
+        accountant = PLDBudgetAccountant(noise=20,
+                                         total_epsilon=1,
+                                         total_delta=1e-5)
+        for _ in range(0, 4):
+            accountant.request_budget(NoiseKind.LAPLACE)
+            accountant.request_budget(NoiseKind.GAUSSIAN)
+
+        optimal_noise = accountant.optimize_noise_to_delta()
```

Comment:
This is the problematic test; I have tried adjusting parameters to make it faster, but it is still very slow 😿